src/huggingface_hub/hf_api.py

```diff
@@ -108,16 +111,51 @@ def logout(self, token: str) -> None:
         r = requests.post(path, headers={"authorization": "Bearer {}".format(token)})
         r.raise_for_status()
 
-    def model_list(self) -> List[ModelInfo]:
+    def list_models(self, filter: Optional[str] = None) -> List[ModelInfo]:
```

Comment:
Fantastic to have the filter option here. This unblocks quite a few use cases in integrations.

Reply:
note that this implem is quite basic, but we could do fancier stuff in the future if needed, like:
- filtering on an array of tags
- exposing "facets" of specialized tags like the ones exposed in the UI at https://huggingface.co/models
- also expose datasets and upcoming other repo types (`spaces`)

cc'ing @Pierrci